File: 62. Unique Paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        if(m<n)
            swap(m,n);
        double res=1.0;
        for(int i=1;i<n;i++)
            res=res*(m-1+i)/i;
        return (int)res;
    }
};

File: 62. Unique paths.js
var uniquePaths = function(m, n,memo = {}) {
    let key = m + ',' + n;
    if(key in memo)
        return memo[key];
    
    
    if(m===0 || n===0)
        return 0;
    
    if(m===1 || n===1)
        return 1;
    
    memo[key] = uniquePaths(m, n-1,memo) + uniquePaths(m-1, n,memo)
    return memo[key] ;
};
